modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -573,6 +578,23 @@ public void endStructDef(String identifier) {
         this.varListStack.pop().forEach(structNode::addField);
         this.compUnit.addTopLevelNode(structNode);
     }
+
+    public void startEnumDef() {
+        this.enumStack.add(TreeBuilder.createEnumNode());
+    }
+
+    public void endEnumDef(String identifier) {
+        EnumNode enumNode = this.enumStack.pop();
+        enumNode.setName(this.createIdentifier(identifier));
+        this.identifierNodeListStack.pop().forEach(identifierNode -> enumNode.addEnumField(identifierNode));
+        this.compUnit.addTopLevelNode(enumNode);
+    }
+
+    public void addEnumFieldList(List<String> enumFieldList) {
+        List<IdentifierNode> identifierList = new ArrayList<>();
```

Comment:
Do we need a Stack of List of identifiers here?

Reply:
+1